GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRModelSceneObject.java

```diff
@@ -43,6 +48,49 @@ public GVRModelSceneObject(GVRContext gvrContext) {
     }
 
     /**
+     * Loads the specified model as a child of this scene object.
+     *
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param pathName
+     *            path to asset to load (from assets, sdcard or URL)
+     * @see GVRAssetLoader.loadModel
+     */
+    public GVRModelSceneObject(GVRContext gvrContext, String pathName) {
+        super(gvrContext);
+        mFileName = pathName;
+        mAnimations = new ArrayList<GVRAnimation>();
+    }
+
+    /**
+     * Gets the filename of the asset associated with the model.
+     * @return model file name or null if none specified
+     */
+    public String getFileName() { return mFileName; }
+
+    /**
+     * Loads the asset specified by the file name provided.
+     * The function returns immediately even though parts of
+     * the asset may still be loading in the background.
+     * It does not add the model to the current scene.
+     * If the model is already in the scene it will be
+     * disabled until all of the textures are available.
+     * If no file name was given on the constructor, this
+     * function does nothing. If the asset has already
+     * been loaded, this function does nothing.
+     * @throws IOException if the asset does not exist
+     * @see GVRModelSceneObject
+     * @see GVRModelSceneObject.getFileName
+     */
+    public void load() throws IOException
```

Comment:
Why do we need to expose yet another way of loading a model? Why don't we have all the loading go through the assetManager?